app/controllers/chatfuel_controller.rb

```diff
@@ -7,14 +7,83 @@ def show
     if tc
       text = Text::Renderer.new(text_component: tc).render(:main_part)
 
-      @response = {
-        messages: [
-          { text: text.first(640)} # Messages for Facebook Messenger can only be 640 characters long. Source: https://developers.facebook.com/docs/messenger-platform/send-api-reference#request
-        ]
-      }
-      render json: @response
+      qa = tc.question_answers
+
+      #Test API Request: http://localhost:3000/chatfuel/1
+      if qa.exists?(1)
+          sendJsonResponse(text, 1, qa.find(1).question)
+      else
+          sendJsonResponse(text)
+      end
+
     else
-      render json: {}, status: 404
+     render json: {}, status: 404
     end
   end
+
+  def showQuestionAndAnswer
+      # request for qa for a specific text component
+      # send answer as json response to chatfuel
+      # if more questions and answers available, append button for next answer to json response
+
+      chatbot_channel = Channel.chatbot
+
+      tc = Text::Sorter.sort(chatbot_channel.text_components, {}).first
+
+      if tc
+          qaId = params[:id]
+
+          if tc.question_answers.exists?(qaId)
+              qa = tc.question_answers.find(qaId)
+
+              if tc.question_answers.exists?(qaId.to_i + 1)
+                  # send answer + button for next question
+                  sendJsonResponse(qa.answer, qaId.to_i + 1, tc.question_answers.find(qaId.to_i + 1).question)
+              else
+                  # send only answer for this question
+                  sendJsonResponse(qa.answer)
+              end
+          else
+              render json: {}, status: 404
+          end
+
+      else
+          render json: {}, status: 404
+      end
+  end
+
+  def sendJsonResponse(text, qaId = 0, questionText = '')
```

Comment:
Add at least a unit test for `#sendJsonResponse` and a controller test for `#showQuestionAndAnswer` (it's a controller action).